### PR TITLE
chore: move storage browser icon data from icon element to react icon primitive

### PR DESCRIPTION
**Pull request by jordanvn**
#### Description of changes
* Move icon paths and variant types to react package icon primitive, to resolve type issue

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by cshfang — DISMISSED** [review]

File: packages/react/src/primitives/Icon/internal.ts [hunk 1]
```diff
@@ -1,2 +1,6 @@
 export * from './icons';
 export { useIcons } from './context/useIcons';
+export {
+  STORAGE_BROWSER_ICON_PATHS,
+  StorageBrowserIconType,
+} from './context/StorageBrowserIcons';
```
Comment (calebpollman) [inline packages/react/src/primitives/Icon/internal.ts:6]: Think this line is creating the public export

File: packages/react/__tests__/__snapshots__/exports.ts.snap [hunk 2]
```diff
@@ -111,6 +111,7 @@ exports[`@aws-amplify/ui-react/internal exports should match snapshot 1`] = `
   "IconVisibilityOff",
   "IconWarning",
   "PrimitiveCatalog",
+  "STORAGE_BROWSER_ICON_PATHS",
```
Comment (calebpollman) [inline packages/react/__tests__/__snapshots__/exports.ts.snap:114]: Should not be a public export, needs to be exported from _packages/react/src/internal.ts_

Comment (calebpollman) [inline packages/react/__tests__/__snapshots__/exports.ts.snap:114]: Should not be a public export

Comment (jordanvn) [reply packages/react/__tests__/__snapshots__/exports.ts.snap]: Removed from snapshot after changing export location

File: packages/react/src/primitives/Icon/internal.ts (hunk 1 above)
Comment (jordanvn) [reply packages/react/src/primitives/Icon/internal.ts]: Added export at suggested file

File: packages/react/__tests__/__snapshots__/exports.ts.snap (hunk 2 above)
Comment (jordanvn) [reply packages/react/__tests__/__snapshots__/exports.ts.snap]: I had thought that I saw this test pass, but it is failing again. It seems like this test is confirming internal exports though, so I think it being in this snapshot should be correct? Or am I missing something?

**Review by calebpollman — APPROVED** [review]

**Pull request merged**